src/Stratis.Bitcoin.Features.Consensus/PosConsensusFeature.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Builder.Feature;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+
+[assembly: InternalsVisibleTo("Stratis.Bitcoin.Features.Miner.Tests")]
+[assembly: InternalsVisibleTo("Stratis.Bitcoin.Features.Consensus.Tests")]
+
+namespace Stratis.Bitcoin.Features.Consensus
+{
+    public class PosConsensusFeature : FullNodeFeature
+    {
+        private readonly IChainState chainState;
+        private readonly IConnectionManager connectionManager;
+        private readonly IConsensusManager consensusManager;
+        private readonly NodeDeployments nodeDeployments;
+        private readonly ConcurrentChain chain;
+        private readonly IInitialBlockDownloadState initialBlockDownloadState;
+        private readonly IPeerBanning peerBanning;
+        private readonly ILoggerFactory loggerFactory;
+
+        public PosConsensusFeature(
+            Network network,
+            IChainState chainState,
+            IConnectionManager connectionManager,
+            IConsensusManager consensusManager,
+            NodeDeployments nodeDeployments,
+            ConcurrentChain chain,
+            IInitialBlockDownloadState initialBlockDownloadState,
+            IPeerBanning peerBanning,
+            ILoggerFactory loggerFactory)
+        {
+            this.chainState = chainState;
+            this.connectionManager = connectionManager;
+            this.consensusManager = consensusManager;
+            this.nodeDeployments = nodeDeployments;
+            this.chain = chain;
+            this.initialBlockDownloadState = initialBlockDownloadState;
+            this.peerBanning = peerBanning;
+            this.loggerFactory = loggerFactory;
+
+            this.chainState.MaxReorgLength = network.Consensus.MaxReorgLength;
+        }
+
+        /// <inheritdoc />
+        public override Task InitializeAsync()
+        {
+            DeploymentFlags flags = this.nodeDeployments.GetFlags(this.consensusManager.Tip);
+            if (flags.ScriptFlags.HasFlag(ScriptVerify.Witness))
```

Comment:
We can now potentially remove witness discovery (this will anyway be false on stratis)

Hmm well lets let @quantumagi do that when he refactors `Deployment`